plugins/janus_sip.c

```diff
@@ -4521,26 +4504,25 @@ static void *janus_sip_handler(void *data) {
 					}
 				}
 			} else if(!strcasecmp(action_text, "pause")) {
-				gint pause_bitmap = ~recorder_bitmap & update_bitmap;
-				if(pause_bitmap & 1)
+				if(record_audio)
 					janus_recorder_pause(session->arc);
-				if(pause_bitmap & 2)
+				if(record_video)
 					janus_recorder_pause(session->vrc);
-				if(pause_bitmap & 4)
+				if(record_peer_audio)
 					janus_recorder_pause(session->arc_peer);
-				if(pause_bitmap & 8)
+				if(record_peer_video)
 					janus_recorder_pause(session->vrc_peer);
 			} else if(!strcasecmp(action_text, "resume")) {
-				gint resume_bitmap = recorder_bitmap & update_bitmap;
-				if(resume_bitmap & 1)
+				gboolean send_pli = FALSE;
+				if(record_audio)
 					janus_recorder_resume(session->arc);
-				if(resume_bitmap & 2)
-					janus_recorder_resume(session->vrc);
-				if(resume_bitmap & 4)
+				if(record_video && !janus_recorder_resume(session->vrc))
+					send_pli = TRUE;
+				if(record_peer_audio)
 					janus_recorder_resume(session->arc_peer);
-				if(resume_bitmap & 8)
-					janus_recorder_resume(session->vrc_peer);
-				if(resume_bitmap & 10)
+				if(record_peer_video && !janus_recorder_resume(session->vrc_peer))
+					send_pli = TRUE;
```

Comment:
You don't need `send_pli` here: if we're recording the peer, we need a PLI from the SIP user, not the WebRTC user. I think you can just do a

	if(record_video && !janus_recorder_resume(session->vrc))
		gateway->send_pli(session->handle);

without the additional `send_pli` boolean definition.